pkg/sentry/control/fs.go

```diff
@@ -50,8 +81,14 @@ func (f *Fs) TarRootfsUpperLayer(o *TarRootfsUpperLayerOpts, _ *struct{}) error
 	defer outFD.Close()
 
 	ctx := f.Kernel.SupervisorContext()
-	mns := f.Kernel.GlobalInit().Leader().MountNamespace()
-	root := mns.Root(ctx)
+	mntns, err := f.mountNamespaceForContainer(o.ContainerID)
+	if err != nil {
+		return err
+	}
+	defer mntns.DecRef(ctx)
+
+	root := mntns.Root(ctx)
+	defer root.DecRef(ctx)
```

Comment:
Ah thanks for fixing this bug! I think we were leaking `mntns.Root(ctx)` ref earlier.